### PR TITLE
Kit Hardware Docs

### PR DESCRIPTION
**Pull request by trickeydan**

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sphinx
+sphinx-markdown-tables
+recommonmark
+sphinx-rtd-theme
```
Comment (kierdavis) [inline docs/requirements.txt]: Is this file needed if we have the packages listed in `pyproject.toml`?

File: docs/source/kit/index.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+`USB B Cable <https://www.amazon.co.uk/-/dp/B00NH11KIK>`__      1
+`Large CamCon <http://uk.farnell.com/-/-/-/dp/3882275>`__       2        7.5mm
+`Medium CamCon <http://uk.farnell.com/-/-/-/dp/3881854>`__      2        5mm
+`Small CamCon <http://uk.farnell.com/-/-/-/dp/1717047>`__       3        3.81mm
+=============================================================== ======== =============
+ 
+.. |Motor Board| image:: /_static/kit/mcv4b.png
+.. |Power Board| image:: /_static/kit/pbv4.png
+.. |Servo Board| image:: /_static/kit/sbv4.png
+.. |Arduino| image:: /_static/kit/arduino.png
```
Comment (kierdavis) [inline docs/source/kit/index.rst]: `arduino.png` is a picture of legacy servo assembly. I suggest deleting it to prevent future confusion.., and renaming `arduino_headers.png` to `arduino.png`.

File: docs/source/kit/arduino.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+The reset button allows you to instantly reboot the Arduino in case it
+isn’t working. This is not a guaranteed fix, but may solve some
+problems.
+
+GPIO Pins
+---------
+
+The Arduino allows you to connect your kit to your own electronics. It has fourteen digital I/O pins, and six analogue. The analogue pins can read an analogue signal from 0 to 5V. The board also has a couple of ground pins, as well as some pins fixed at 3.3V and 5V output.
+
+The pin layout of the servo assembly is the same as an Arduino Uno.
```
Comment (kierdavis) [inline docs/source/kit/arduino.rst]: The servo assembly isn't relevant anymore. I don't think this line is needed at all now.

File: docs/source/kit/arduino.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+.. Warning:: Ultrasound should only be considered accurate up to around two metres, beyond which the signal can become distorted and produce erroneous results.
+
+The sensor has four pin connections: ground, 5V (sometimes labelled
+*vcc*), *trigger* and *echo*. Most ultrasound sensors will label which
+pin is which. The ground and 5V should be wired to the ground and 5V
+pins of the Arduino respectively. The trigger and echo pins should be
+attached to two different digital IO pins. Take note of these two pins,
+you’ll need them to use the sensor.
+
+.. Hint:: If the sensor always returns a distance of zero, it means the *trigger* and *echo* pins are connected the wrong way! Either change the pin numbers in the code, or swap the connections.
```
Comment (kierdavis) [inline docs/source/kit/arduino.rst]: There's other reasons why a sensor might return a distance of zero.

```suggestion
.. Hint:: If the sensor always returns a distance of zero, it might mean the *trigger* and *echo* pins are connected the wrong way! Either change the pin numbers in the code, or swap the connections.
```

Also see also #16.

File: docs/source/kit/arduino.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+Designs
+-------
+
+The schematic diagrams for the Arduino is below, as
+well as the source code of the firmware on the Arduino. You do not need
+this information to use the board but it may be of interest to some
+people.
+
+-  `Arduino Uno
+   Schematic <https://www.arduino.cc/en/uploads/Main/Arduino_Uno_Rev3-schematic.pdf>`__
```
Comment (kierdavis) [inline docs/source/kit/arduino.rst]: Is it worth hosting a copy of this ourselves?

File: docs/source/kit/battery.rst [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+.. Warning:: You must not use any batteries, chargers, bags or cables not explicitly authorised by SourceBots.
+
+Warnings
+--------
+
+-  Never leave batteries unattended when they are in use or charging.
+-  Always place the batteries in the provided charging bag when charging
+   or storing.
+-  Always follow the charging checklist precisely to make sure the
```
Comment (kierdavis) [inline docs/source/kit/battery.rst]: Not needed since competitors will not be charging batteries,

File: docs/source/kit/battery.rst [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+   charger is correctly configured.
+-  Always leave the charging leads connected to the battery charger at
```
Comment (kierdavis) [inline docs/source/kit/battery.rst]: Not needed since competitors will not be charging batteries,

File: docs/source/kit/battery.rst [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+battery to be placed in, so that accidental collisions do not damage the
+battery.
+
+Flat batteries
+--------------
+
+When the battery has been almost completely discharged, the Power Board
+will automatically turn off and the LED marked “Power / Flat Battery
+Indicator” in the `diagram </kit/power-board>`__ will
+flash red and green. You should immediately disconnect the battery.
```
Comment (kierdavis) [inline docs/source/kit/battery.rst]: Probably worth saying that the competitor should then ask a faciliator to exchange their flat battery for a fresh one, to avoid any ambiguity around whether they should be trying to charge it themselves or not.

File: docs/source/kit/motor-board.rst [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
++---------------+--------------------------------------+---------------+
+| LED           | Meaning                              | Initial       |
+|               |                                      | power-up      |
+|               |                                      | state         |
++===============+======================================+===============+
+| Power         | The board is powered                 | On            |
++---------------+--------------------------------------+---------------+
+| M0/M1         | Brightness indicates speed, colour   | Off           |
+| Speed/Directi | indicates direction                  |               |
```
Comment (kierdavis) [inline docs/source/kit/motor-board.rst]: Breaking the word "Direction" across two lines here results in there being an unsightly space in the middle of the word in the rendered page: https://sbot.readthedocs.io/en/latest/kit/motor-board.html#indicators

File: docs/source/kit/pi.rst [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+.. Warning:: The model of Pi you have will make no difference to the functioning of your robot.
+
+Technical Details
+-----------------
+
+Your robot is running a customised version of the Raspbian_ operating system.
+
+When a USB stick is inserted, the SourceBots software will look for a *main.py*, and then execute it.
+
+The output of your code is logged to the *systemd journal*, and also written to a file on the USB stick.
```
Comment (kierdavis) [inline docs/source/kit/pi.rst]: Let's sort this by order of relevance to a competitor.

```suggestion
The output of your code is written to a file on the USB stick, and also logged to the *systemd journal*.
```

File: docs/source/kit/power-board.rst [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+   Power Board Diagram
+
+Connectors
+----------
+
+There are six power output connectors on the board, labelled L0–L3, H0,
+and H1. These can supply around 11.1V (±15%). The “H” connectors will
+supply more current than the “L” connectors.
+
+They should be used to connect to the motor board power input, though
```
Comment (kierdavis) [inline docs/source/kit/power-board.rst]: and servo board

File: docs/source/kit/power-board.rst [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+can also be used to power other devices. These are enabled when your
+robot code is started and can also be turned on or off from your code.
+
+There are two 5V connectors that can be used to connect low-current
+devices that take 5V inputs, such as the Raspberry Pi and the servo
```
Comment (kierdavis) [inline docs/source/kit/power-board.rst]: Servo shield isn't used anymore.

File: docs/source/kit/power-board.rst [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+
+Indicators
+----------
+
++---------+-----------------------------------------------------+----------+
+| LED     | Meaning                                             | Initial  |
+|         |                                                     | power-up |
+|         |                                                     | state    |
++=========+=====================================================+==========+
+| PWR |   | Green when poweredFlashing red and green when the   | Green    |
```
Comment (kierdavis) [inline docs/source/kit/power-board.rst]: Weird formatting in the "Meaning" column here.

File: docs/source/kit/power-board.rst [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+| RUN |   | Orange on power-up, or USB reset Flashing green     | Orange   |
+| ERROR   | when ready to run Solid green when running or       |          |
+|         | booting                                             |          |
++---------+-----------------------------------------------------+----------+
+
+On power-up, the Power Board will emit some beeps, which are related to
+the version of the firmware it has installed.
+
+If the Power Board starts beeping (and all the outputs turn off) then
+this means that the whole board’s current limit has been triggered.
```
Comment (kierdavis) [inline docs/source/kit/power-board.rst]: Weird apostrophe.

```suggestion
this means that the whole board's current limit has been triggered.
```

File: docs/source/kit/pi.rst [hunk 15]
```diff
@@ -0,0 +1,25 @@
+Raspberry Pi
+============
+
+.. DANGER:: Do not reflash or edit the SD card. This will stop your robot working!
+
+.. figure:: /_static/kit/pi2.jpg
```
Comment (kierdavis) [inline docs/source/kit/pi.rst]: There is both a `pi.jpg`, used in `docs/source/kit/index.rst`, and a `pi2.jpg`, used here. Do we really need both?

File: docs/source/kit/servo-board.rst [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+
+   Board Diagram
+
+Indicators
+----------
+
+===== ==================== ======================
+LED   Meaning              Initial power-up state
+===== ==================== ======================
+Power The board is powered On
```
Comment (kierdavis) [inline docs/source/kit/servo-board.rst]: This table is incomplete, but updating it (and likely the diagram too) is an upstream issue: srobo/docs#103. I have opened #17 to track fixing this in this repo once the upstream issue is resolved.

File: docs/requirements.txt (hunk 1 above)
Comment (trickeydan) [reply docs/requirements.txt]: It's needed for readthedocs unfortunately.

File: docs/source/kit/pi.rst (hunk 15 above)
Comment (trickeydan) [reply docs/source/kit/pi.rst]: `pi2.jpg` is a much better image. I suggest removing `pi.jpg`

Comment (kierdavis) [reply docs/source/kit/pi.rst]: Added to #18.

**Comment by trickeydan** [comment]
> Generally good, though as usual I come equipped with a large bag of nitpicks.
> 
> We should really add attribution for photos we used. I'd suggest we even include it for ones we've taken ourselves, so that other people can replicate it if they reuse the photo elsewhere.
> 
>     * [ ]  `arduino_headers.png`: SourceBots (assuming this is the photo you took yesterday)
> 
>     * [ ]  `arduino_pinout.png`: attribution already included in image
> 
>     * [ ]  `arduino.png`: SourceBots (note that one of my other comments suggests deleting this anyway)
> 
>     * [ ]  `battery.jpg`: ???
> 
>     * [ ]  `mcv4b.png`: Student Robotics
> 
>     * [ ]  `mcv4b_diagram.png`: Student Robotics
> 
>     * [ ]  `pbv4.png`: Student Robotics
> 
>     * [ ]  `pbv4_diagram.png`: Student Robotics
> 
>     * [ ]  `pi.jpg`: ???
> 
>     * [ ]  `pi2.jpg`: ???
> 
>     * [ ]  `sbv4.png`: Student Robotics
> 
>     * [ ]  `sbv4_diagram.png`: Student Robotics

I plan to replace most of the images later so that the boards etc have cases on them. The exception to this is the diagrams, which I will check the authors of upstream.

File: docs/source/kit/index.rst (hunk 2 above)
Comment (trickeydan) [reply docs/source/kit/index.rst]: To do in #18

File: docs/source/kit/motor-board.rst (hunk 9 above)
Comment (trickeydan) [reply docs/source/kit/motor-board.rst]: Blame pandoc :angry: Fixed

**Comment by kierdavis** [comment]
There's one comment left on `docs/source/kit/power-board.rst` that should really be addressed; other than that, I'm happy with this.
